app/views/analysis/index.html.erb

```diff
@@ -0,0 +1,624 @@
+    return get_axis_values(min_y_value, max_y_value, 2);
+  }
+  else if(y_difference >=12 && y_difference <= 30){
+    return get_axis_values(min_y_value, max_y_value, 4);
+  }
+  else if(y_difference >= 31 && y_difference <= 50){
+    return get_axis_values(min_y_value, max_y_value, 6);
+  }
+  else if(y_difference >= 51 && y_difference <= 70){
+    return get_axis_values(min_y_value, max_y_value, 8);
+  }
+  else if(y_difference >= 71 && y_difference <= 89){
+    return get_axis_values(min_y_value, max_y_value, 12);
+  }
+  else if(y_difference >= 90){
+    return get_axis_values(min_y_value, max_y_value, 15);
+  }
+  else if(y_difference >= 100 && y_difference <= 200){
+    return get_axis_values(min_y_value, max_y_value, 25);
+  }
+  else if(y_difference >= 201 && y_difference <= 500){
+    return get_axis_values(min_y_value, max_y_value, 50);
+  }
+  else if(y_difference >= 501 && y_difference <= 999){
+    return get_axis_values(min_y_value, max_y_value, 100);
+  }
+  else if(y_difference >= 1000 && y_difference <= 2000){
+    return get_axis_values(min_y_value, max_y_value, 250);
+  }
+  else if(y_difference >= 2001 && y_difference <= 5000){
+    return get_axis_values(min_y_value, max_y_value, 500 );
+  }
+  else if(y_difference >= 5001 && y_difference <= 10000){
+    return get_axis_values(min_y_value, max_y_value, 1000 );
+  }
+  else if(y_difference >= 10001 && y_difference <= 50000){
+    return get_axis_values(min_y_value, max_y_value, 2000 );
+  }
+  else if(y_difference >= 50001 && y_difference <= 99999){
+    return get_axis_values(min_y_value, max_y_value, 4000 );
+  }
+  else if(y_difference >= 100000 && y_difference <= 499999){
+    return get_axis_values(min_y_value, max_y_value, 40000);
+  }
+  else if(y_difference >= 500000 && y_difference <= 1000000){
+    return get_axis_values(min_y_value, max_y_value, 70000);
+  }
+}
+
+
+// Function that determines if a given year is valid.
+function check_if_valid(id, min_val)
+{
+  if(id.indexOf(".") == -1 && id.indexOf("-") == -1){
+    term = parseInt(id);
+    if(term >= min_val){
+      return true;
+    }
+    return false;
+  }
+  return false;
+}
+function common_deaths_age() {
+  // Default configurations when page is loaded by the user.
+  $("#demo").html("");
+  var sent_input =
+  {"sex": [],
+  "race" : [],
+  "start_year" : "1985",
+  "end_year" : "2017",
+  "num_causes" : "20",
+  "cause_of_death" : "",
+  "age_range" : [],
+  "specified_criteria" : [],
+  "residence_zipcode" : "",
+  "death_zipcode" : ""
+}
+  var filled_input = showValues();
+  for(var i in filled_input)
+  {
+    if(filled_input[i] != ""){
+      sent_input[i] = filled_input[i]
+    }
+  }
+
+  for(var j in sent_input)
+  {
+    if(Array.isArray(sent_input[j]))
+    {
+      sent_input[j] = JSON.stringify(sent_input[j]);
+    }
+  }
+  if(!check_if_valid(sent_input["start_year"], 1900)){
+    $("#demo").html("Invalid year entered! Starting year must be from 1900 and onwards!");
```

Comment:
Yeah, it will clear it since the first line of the function is $("#demo").html("");